### PR TITLE
Moved counter logic to new file. Added unit test file for mileage counter.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: app/src/main/java/com/gsandchase/mileagelogger/Logger.java [hunk 1]
```diff
@@ -1,81 +0,0 @@
-package com.gsandchase.mileagelogger;
```
Comment (minorhacks) [inline app/src/main/java/com/gsandchase/mileagelogger/Logger.java:1]: This class seems unnecessary - Log.d()/Log.v() etc. can be used for debugging purposes. They write to a systemwide logfile that can be examined with logcat.

File: app/src/main/java/com/gsandchase/mileagelogger/MainActivity.java [hunk 2]
```diff
@@ -30,13 +30,10 @@
     TextView totalMileage;
     Button addMileageButton;
     EditText newMileage;
-    Context context;
 
-    private float weeklyMileage = 0, monthlyMileage = 0, annualMileage = 0;
```
Comment (minorhacks) [inline app/src/main/java/com/gsandchase/mileagelogger/MainActivity.java:35]: I want to remove as much state from the class as makes sense. The app contains an object that is responsible for keeping track of mileage totals, which replaces all of these data members.

File: app/src/main/java/com/gsandchase/mileagelogger/MainActivity.java [hunk 3]
```diff
@@ -50,24 +47,18 @@ protected void onCreate(Bundle savedInstanceState) {
...
-            weeklyMileage = 0;
-            weeklyMileageS = "0";
-        }
 
-        totalMileage.setText(weeklyMileageS);
+        updateCounterText();
 
 
         addMileageButton.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
```
Comment (minorhacks) [inline app/src/main/java/com/gsandchase/mileagelogger/MainActivity.java:58]: Changed the onClick handler to do two things: update the mileage counter internally, and trigger an update of the on-screen counter.

File: app/src/main/java/com/gsandchase/mileagelogger/MileageCounter.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
+package com.gsandchase.mileagelogger;
```
Comment (minorhacks) [inline app/src/main/java/com/gsandchase/mileagelogger/MileageCounter.java:1]: This class keeps track of miles for each day. From there, it's easy to total up the miles for a given week/month/year. (I implemented the one for weeks - I'll leave the month/year for you!). You can also add methods to this class to serialize/deserialize to a file.

File: app/src/test/java/com/gsandchase/mileagelogger/MileageCounterTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import org.junit.Test;
+
+import java.util.Date;
+
+import static org.junit.Assert.*;
+
+/**
+ * To work on unit tests, switch the Test Artifact in the Build Variants view.
+ */
+public class MileageCounterTest {
```
Comment (minorhacks) [inline app/src/test/java/com/gsandchase/mileagelogger/MileageCounterTest.java:12]: Some example tests. Writing the tests for a class FIRST and then implementing the method ensures that you create the API that is most pleasant to use. See https://www.youtube.com/watch?v=pqomi6W4AJ4

File: app/src/main/java/com/gsandchase/mileagelogger/MileageCounter.java (hunk 4 above)
Comment (rusty2123) [inline app/src/main/java/com/gsandchase/mileagelogger/MileageCounter.java:1]: I will implement those eventually, but for now I'd just like a working app for weekly mileage. I also don't know what serializing a file is.

**Pull request merged**